In [25]:
import pandas as pd
import pyodbc as odbc
import io

 Define database connection details of ms sql server

In [27]:
server = "ichchha\SQLEXPRESS"
database = "pro2"
username = "ICHCHHA/WELCOME"
password = "Jigyasa1701*"
DRIVER_NAME = "SQL SERVER"


In [9]:
table_name = "used_bikes"

1-Read the csv or excel file

In [10]:
data =pd.read_csv(r"E:\dddd\used_bikes.csv")

In [13]:
# memory usage before
total_mem_before = data.info(memory_usage="deep")


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 149 entries, 0 to 148
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   bike_name   149 non-null    object 
 1   price       149 non-null    float64
 2   city        149 non-null    object 
 3   kms_driven  149 non-null    float64
 4   owner       149 non-null    object 
 5   age         149 non-null    float64
 6   power       149 non-null    float64
 7   brand       149 non-null    object 
dtypes: float64(4), object(4)
memory usage: 45.0 KB


memory usage before: 45.0 KB

In [14]:
# Change the datatypes of the column and check the memory usage before and after the change in the data types

data["city"] = data["city"].astype("category")
data["owner"] = data["owner"].astype("category")
data["brand"] = data["brand"].astype("category")
total_mem_after = data.info(memory_usage="deep")


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 149 entries, 0 to 148
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype   
---  ------      --------------  -----   
 0   bike_name   149 non-null    object  
 1   price       149 non-null    float64 
 2   city        149 non-null    category
 3   kms_driven  149 non-null    float64 
 4   owner       149 non-null    category
 5   age         149 non-null    float64 
 6   power       149 non-null    float64 
 7   brand       149 non-null    category
dtypes: category(3), float64(4), object(1)
memory usage: 22.2 KB


memory usage after: 22.2 KB

3-Dump the data into the mysql database

In [31]:
# Connect to SQL Server
connection_string = f""" 
DRIVER={{{DRIVER_NAME}}};
SERVER={"server"};
DATABASE={database};
TRUST_CONNECTION = YES;
"""
CONN =odbc.connect=(connection_string)






In [40]:
drop_index_sql = f"DROP INDEX IF EXISTS idx_{table_name}"


In [ ]:
#Execute drop index query
DROP_INDEX = CONN.DROP_INDEX ()
DROP_INDEX .execute(drop_index_sql)
CONN.commit()


In [ ]:
create_index_sql = f"CREATE INDEX idx_{table_name} ON {table_name}(city, owner, brand)"
cursor.execute(create_index_sql)
conn.commit()

# Close connection
conn.close()

print(f"Data successfully dumped and indexed in table '{table_name}'")